Firestore/Example/Tests/Local/FSTLocalSerializerTests.mm

```diff
@@ -162,7 +162,7 @@ - (void)testEncodesDeletedDocumentAsMaybeDocument {
 - (void)testEncodesQueryData {
   FSTQuery *query = FSTTestQuery("room");
   FSTTargetID targetID = 42;
-  FSTSnapshotVersion *version = FSTTestVersion(1039);
+  const SnapshotVersion version = testutil::Version(1039);
```

Comment:
I don't see any great value in marking these local variables `const`. It's difficult to see the difference between a const copy and a const reference:
```c++
const SnapshotVersion version = other;   // copy
const SnapshotVersion& version = other;  // a reference to the other instance
```
The benefit of marking your locals const is that you don't accidentally invoke mutating methods when you don't mean to. For the most part though, we don't have mutating methods: we prize immutable types to avoid this problem.

Meanwhile, too much copying is also a plague of C++ programs and the extra `const` decoration makes this much harder to see.

Especially for immutable types, I propose we drop `const` from locals when we've copied:
```c++
SnapshotVersion version = other;        // copy is much more obvious
const SnapshotVersion& version = other; // a reference to somebody else's version.
```